core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.js

```diff
@@ -0,0 +1,231 @@
+const LABEL_FOR_SUBMIT_BUTTON = 'Submit and start contributing';
+
+module.exports = class puppeteerUtilities {
+  page;
+  browserObject;
+  userHasAcceptedCookies = false;
+
+  /**
+   * This is a function that opens a new browser instance for the user.
+   * @returns {Promise<puppeteer.Page>} - Returns a promise that resolves
+   * to a Page object controlled by Puppeteer.
+   */
+  async openBrowser() {
+    /* Currently, headless is set to false and the page viewport
+       is maximized so that it would be easy for the developers
+       to debug easily while testing.
+       We can remove these settings before merging as we have
+       to run the tests in headless mode. */
+    await puppeteer
+      .launch({
+        headless: false,
+        args: ['--start-fullscreen', '--use-fake-ui-for-media-stream']
+      })
+      .then(async(browser) => {
+        this.browserObject = browser;
+        this.page = await browser.newPage();
+        await this.page.setViewport({ width: 0, height: 0 });
+        // Accepting the alerts that appear in between the tests.
+        await this.page.on('dialog', async dialog => {
+          await dialog.accept();
+        });
+      });
+
+    return this.page;
+  }
+
+  /**
+   * Function to sign in the user with the given email to the Oppia website.
+   * @param {string} email - The email of the user.
+   */
+  async signInWithEmail(email) {
+    await this.goto(testConstants.URLs.home);
+    if (!this.userHasAcceptedCookies) {
+      await this.clickOn('OK');
+      this.userHasAcceptedCookies = true;
+    }
+    await this.clickOn('Sign in');
+    await this.type(testConstants.SignInDetails.inputField, email);
+    await this.clickOn('Sign In');
+    await this.page.waitForNavigation({waitUntil: 'networkidle0'});
+  }
+
+  /**
+   * This function signs up a new user with the given username and email.
+   * @param {string} userName - The username of the user.
+   * @param {string} signInEmail - The email of the user.
+   */
+  async signUpNewUser(userName, signInEmail) {
+    await this.signInWithEmail(signInEmail);
+    await this.type('input.e2e-test-username-input', userName);
+    await this.clickOn('input.e2e-test-agree-to-terms-checkbox');
+    await this.page.waitForSelector(
+      'button.e2e-test-register-user:not([disabled])');
+    await this.clickOn(LABEL_FOR_SUBMIT_BUTTON);
+    await this.page.waitForNavigation({waitUntil: 'networkidle0'});
+  }
+
+  /**
+   * Function to reload the current page.
+   */
+  async reloadPage() {
+    await this.page.reload({waitUntil: ['networkidle0', 'domcontentloaded']});
+  }
+
+  /**
+   * The function clicks the element using the text on the button.
+   * @param {string} selector - The text on the button or the CSS selector of
+   * the element to be clicked
+   */
+  async clickOn(selector) {
+    try {
+      const [button] = await this.page.$x(
+        '//' + '*[contains(text(), "' + selector + '")]');
```

Comment:
yes, but some thing like this: ``//` + `*[contains(text(), "${selector}")]``. we need // separate and the following string seperate as // is impying that string as comment, so just separating the `//` and ``*[contains(text(), "${selector}")]``

Reply:
I don't understand you explanation.

Reply:
I have changed this to `'//' + `*[contains(text(), "${selector}")]``. but it cannot be changed to ``//*[contains(text(), "${selector}")]`` because // is starting a comment, so thats why we need to separate it.